x-pack/plugin/esql/build.gradle

```diff
@@ -113,19 +120,27 @@ tasks.named("test").configure {
           from snippetsFolder
           into snippetsDocFolder
           include '**/*.md'
-          preserve {
-            // The snippets directory contains generated and static content, so we must preserve all MD files.
-            include '**/*.md'
+          if (countSnippets <= 100) {
```

Comment:
We used to delete all files if we ran more than 1 test, which is extreme. This way we only delete all files if we probably run the entire suite.

Reply:
I agree the comment is ambiguous. The second half of the comment is better. Anyway, the way it works is if we run only a subset of the tests (usually just one, hence the old rule), then it uses `include '**/*.md'` to keep all old files. If, instead, we run all the tests, it will try to delete unused files. We used to delete all unused files (too many), but now we delete only the ones we feel more confident are actually generated. The more complex line to handle this is:

```
include '*.md', '**/operators/*.md', '**/operators/**/*.md', '**/lists/*.md'
```

Reply:
I verified this line by running all tests and seeing what got deleted, and verifying that it was nothing we didn't already want to delete.

Reply:
I deleted the `not` in the second comment below, since it was the source of the confusion.